mitiq/pec/tests/test_pec_sampling.py

```diff
@@ -217,67 +221,75 @@ def test_sample_circuit_trivial_decomposition():
-#     """Tests the sample_circuit by comparing the exact Choi matrices."""
-#     ideal_choi = _circuit_to_choi(twoq_circ)
-#     noisy_circuit = twoq_circ.with_noise(cirq.depolarize(BASE_NOISE))
-#     noisy_choi = _circuit_to_choi(noisy_circuit)
-#     choi_unbiased_estimates = []
-#     for _ in range(500):
-#         imp_circuit, sign, norm = sample_circuit(
-#             twoq_circ, decomposition_dict
-#         )
-#         noisy_imp_circuit = imp_circuit.with_noise(
-#             cirq.depolarize(BASE_NOISE)
-#         )
-#         imp_circuit_choi = _circuit_to_choi(noisy_imp_circuit)
-#         choi_unbiased_estimates.append(norm * sign * imp_circuit_choi)
-#     choi_pec_estimate = np.average(choi_unbiased_estimates, axis=0)
-#
-#     noise_error = np.linalg.norm(ideal_choi - noisy_choi)
-#     pec_error = np.linalg.norm(ideal_choi - choi_pec_estimate)
-#     assert pec_error < noise_error
-#     assert np.allclose(ideal_choi, choi_pec_estimate, atol=0.05)
+BASE_NOISE = 0.02
+qreg = cirq.LineQubit.range(2)
+
+
+@pytest.mark.parametrize("gate", [cirq.Y, cirq.CNOT])
+def test_sample_sequence_choi(gate: cirq.Gate):
+    """Tests the sample_sequence by comparing the exact Choi matrices."""
+    qreg = cirq.LineQubit.range(gate.num_qubits())
+    ideal_op = gate.on(*qreg)
+    ideal_circ = cirq.Circuit(ideal_op)
+    noisy_op_tree = (
+        [ideal_op] + [cirq.depolarize(BASE_NOISE)(q) for q in qreg]
+    )
+
+    ideal_choi = _operation_to_choi(ideal_op)
+    noisy_choi = _operation_to_choi(noisy_op_tree)
+
+    representation = local_depolarizing_representation(ideal_circ, BASE_NOISE)
+
+    choi_unbiased_estimates = []
+    rng = np.random.RandomState(1)
+    for _ in range(500):
+        imp_seq, sign, norm = sample_sequence(
+            ideal_circ, [representation], random_state=rng
+        )
+        noisy_sequence = imp_seq.with_noise(cirq.depolarize(BASE_NOISE))
+        sequence_choi = _circuit_to_choi(noisy_sequence)
+        choi_unbiased_estimates.append(norm * sign * sequence_choi)
+
+    choi_pec_estimate = np.average(choi_unbiased_estimates, axis=0)
+    noise_error = np.linalg.norm(ideal_choi - noisy_choi)
+    pec_error = np.linalg.norm(ideal_choi - choi_pec_estimate)
+
+    assert pec_error < noise_error
+    assert np.allclose(ideal_choi, choi_pec_estimate, atol=0.05)
+
+
+def test_sample_circuit_choi():
+    """Tests the sample_circuit by comparing the exact Choi matrices."""
+    # A simple 2-qubit circuit
+    qreg = cirq.LineQubit.range(2)
+    ideal_circ = cirq.Circuit(
+        cirq.X.on(qreg[0]), cirq.I.on(qreg[1]), cirq.CNOT.on(*qreg),
+    )
+
+    noisy_circuit = ideal_circ.with_noise(cirq.depolarize(BASE_NOISE))
+
+    ideal_choi = _circuit_to_choi(ideal_circ)
+    noisy_choi = _operation_to_choi(noisy_circuit)
+
+    rep_list = []
+    for op in ideal_circ.all_operations():
+        rep_list.append(
+            local_depolarizing_representation(cirq.Circuit(op), BASE_NOISE)
+        )
```

Comment:
It would be nice to have a helper function which does these lines then returns `rep_list`. See main review comments.

Reply:
+1